cpp/core/shuffle/ArrowShuffleWriter.cc

```diff
@@ -236,12 +238,29 @@ arrow::Status ArrowShuffleWriter::Split(ColumnarBatch* batch) {
   EVAL_START("split", options_.thread_id)
   ARROW_ASSIGN_OR_RAISE(
       auto rb, arrow::ImportRecordBatch(batch->exportArrowArray().get(), batch->exportArrowSchema().get()));
-  if (schema_ == nullptr) {
-    schema_ = rb->schema();
-    RETURN_NOT_OK(InitColumnType());
+  if (!partitioner_->HasHashValues()) {
```

Comment:
Is this line duplicated with line 253?

Reply:
ArrowFallbackRangeShuffleWriter first column is pid and HashPartitioning first column is hashed key, both of them needs to remove first column

Reply:
HasHashValues()  is to determine whether  the first column is partition key hash values , for partitioner SinglePart and RoundRobin the value is false, for partitioner Range and Hash the value is true. So in line 241 we init schema_ without remove first column, but in line 253 we remove first column and init schema_

Reply:
We should change the naming, maybe `HasPid`? ArrowFallbackRangeShuffleWriter first column is not relevant with `hash`

Reply:
Thanks, I think HasPid is good.